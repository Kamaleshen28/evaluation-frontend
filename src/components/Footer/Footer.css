.footer {
  display: flex;
  padding: 0 240px 0 240px;
  height: 90px;
  align-items: center;
  font-weight: bold;
  color: white;
}

@media only screen and (max-width: 1000px) {
  .footer {
    padding: 0 40px 0 40px;
  }
}

@media only screen and (max-width: 449px) {
  .footer {
    /* height: 180px; */
    flex-direction: column;
    /* justify-content: space-around; */
    padding: 60px 40px 60px 40px
  }
}

@media only screen and (min-width: 450px) {
  .footer {
    justify-content: space-between;

  }
}

.blue-theme {
  background-color: blue;
}

.black-theme {
  background-color: black;
}

.violet-theme {
  background-color: blueviolet;
}

.gray-theme {
  background-color: gray;
}



.theme-text {
  margin-right: 20px;
}

.color-1 {
  width: 50px;
  height: 50px;
  background-color: black;
  border-radius: 5px;
  margin-right: 10px;
}

.color-2 {
  width: 50px;
  height: 50px;
  background-color: blueviolet;
  border-radius: 5px;
  margin-right: 10px;
}

.color-3 {
  width: 50px;
  height: 50px;
  background-color: blue;
  border-radius: 5px;
  margin-right: 10px;
}

.color-4 {
  width: 50px;
  height: 50px;
  background-color: gray;
  border-radius: 5px;
  margin-right: 10px;
}

.footer {
  display: flex;
  justify-content: space-between;
}

.theme-color-container {
  display: flex;
  align-items: center;
}

.save-theme-button-container {
  display: flex;
  align-items: center;
  /* margin-left: 220px; */
}

.save-theme-button {
  height: 30px;
  width: 140px;
  background-color: white;
  color: blue;
  border: none;
  border-radius: 7px;
}