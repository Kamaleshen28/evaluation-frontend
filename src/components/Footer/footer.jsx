/* eslint-disable indent */
import React, { useContext, useEffect, useState } from 'react';
import { eventifyContext } from '../../context/eventifyContext';
import './Footer.css';
import axios from 'axios';

export default function Footer() {
  const { theme, setTheme } = useContext(eventifyContext);
  const [currentThemeId, setCurrentThemeId] = useState({
    id: 0,
    themeSaved: false
  });

  const fetchThemeData = async () => {
    const data = await axios.get('http://localhost:8000/api/themes', {
      headers: {
      }
    });
    (data.data.themes).forEach(eacThemeData => {
      if (eacThemeData.id === data.data.preferredThemeId) {
        setTheme(eacThemeData.colorHexCode);
        setCurrentThemeId(previousData => {
          return { ...previousData, id: eacThemeData.id };
        });
      }
    });
  };
  useEffect(() => {
    fetchThemeData();
  }, []);

  let colorClass = '';
  switch (theme) {
    case '#0000FF': colorClass = 'footer blue-theme'; break;
    case '#000000': colorClass = 'footer black-theme'; break;
    case '#800080': colorClass = 'footer violet-theme'; break;
    case '#9B9999': colorClass = 'footer gray-theme';
  }

  const handleSaveTheme = async () => {
    const data = { 'preferredThemeId': currentThemeId.id };
    const config = {
      method: 'put',
      url: 'http://localhost:8000/api/themes',
      headers: {},
      data: data
    };
    const response = await axios(config);
    setCurrentThemeId(previousData => {
      return { ...previousData, themeSaved: true };
    });
    console.log(response);
  };

  const changeThemeOnClick = (id, currentTheme) => {
    setCurrentThemeId(previousData => {
      return { ...previousData, id: id };
    });
    setTheme(currentTheme);
  };

  return (
    <div className={colorClass}>
      {/* <div className="footer-wrapper"> */}
      <div className="theme-color-container">
        <span className="theme-text">THEMES</span>
        {currentThemeId.id != 1 && <span className="color-1" onClick={() => changeThemeOnClick(1, '#000000')}></span>}
        {currentThemeId.id != 2 && <span className="color-2" onClick={() => changeThemeOnClick(2, '#800080')}></span>}
        {currentThemeId.id != 3 && <span className="color-3" onClick={() => changeThemeOnClick(3, '#0000FF')}></span>}
        {currentThemeId.id != 4 && <span className="color-4" onClick={() => changeThemeOnClick(4, '#9B9999')}></span>}
      </div>

      <div className="save-theme-button-container">
        {!currentThemeId.themeSaved && <button className="save-theme-button" onClick={handleSaveTheme} >SAVE THEME</button>}
      </div>


      {/* </div> */}

    </div>
  );
}