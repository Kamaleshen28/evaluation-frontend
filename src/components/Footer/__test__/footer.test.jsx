import { render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import Footer from '../footer';


jest.mock('react-router-dom');


describe('card', () => {
  it('should render header correctly', async () => {
    const { asFragment } = render(<Footer />);
    await waitFor(() => {
      expect(screen.getAllByText('THEMES')).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});



