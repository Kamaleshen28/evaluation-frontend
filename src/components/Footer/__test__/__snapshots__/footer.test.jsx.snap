// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should render header correctly 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <div
      class="theme-color-container"
    >
      <span
        class="theme-text"
      >
        THEMES
      </span>
      <span
        class="color-1"
      />
      <span
        class="color-2"
      />
      <span
        class="color-3"
      />
    </div>
    <div
      class="save-theme-button-container"
    >
      <button
        class="save-theme-button"
      >
        SAVE THEME
      </button>
    </div>
  </div>
</DocumentFragment>
`;
